src/components/onlineTest/StudentForm/BasicDetails/index.jsx

```diff
@@ -278,6 +287,11 @@ const BasicDetails = ({
           <Controller
             control={control}
             name="MiddleName"
+            rules={{
+              required: true,
+              pattern: /^[\p{L}\p{M}]+$/u,
```

Comment:
This seems to be a good regex to use for multilingual support. Why aren't we using it for the first name as well? Also, we should allow `'` as this may appear in a person's name. For example, D'Souza is a valid last name. Finally, as this pattern will be used for any name validation, which could appear in other places on the site, we may want to export it as a constant from here: https://github.com/navgurukul/kalam/blob/dev/src/utils/constants.js and then import it here. So for example: 
```
  export const INPUT_PATTERNS = { name: /^[\p{L}\p{M}']+$/u; }
```
Then you could reference it here:
```suggestion
              pattern: INPUT_PATTERNS.name,
```
This would allow us to handle all name validation in a uniform way and enable changes to be made to the pattern everywhere by fixing it in one place (e.g., adding the `'` as one of the valid characters in a name would only require an update to the above line in the constants file).